### ISSUE TITLE
String operators not recognized

### ISSUE THREAD

**Issue by nijel** [phpmyadmin/sql-parser#116]
The `ORDER BY` clause accepts expressions, the parser seems to be choking on it.

Failure was originally reported in https://github.com/phpmyadmin/phpmyadmin/issues/4962#issuecomment-270511020

```
./bin/lint-query --query " (select DivId, ClId, TfId, TfName, TfDefault, TfRegExp, TfClasses, DivViewOrder, ClViewOrder from Divisions inner join Classes on DivTournament=ClTournament and DivAthlete=ClAthlete inner join TargetFaces Tf on DivTournament=TfTournament and concat(trim(DivId),trim(ClId)) REGEXP TfRegExp WHERE DivTournament=654 AND DivAthlete='1' AND (ClDivisionsAllowed='' or find_in_set(DivId, ClDivisionsAllowed))) UNION (select DivId, ClId, TfId, TfName, TfDefault, TfRegExp, TfClasses, DivViewOrder, ClViewOrder from Divisions inner join Classes on DivTournament=ClTournament and DivAthlete=ClAthlete inner join TargetFaces Tf on DivTournament=TfTournament and concat(trim(DivId),trim(ClId)) like TfClasses WHERE DivTournament=654 AND DivAthlete='1' AND (ClDivisionsAllowed='' or find_in_set(DivId, ClDivisionsAllowed))) order by DivViewOrder , ClViewOrder , TfDefault desc , TfRegExp>'' desc , concat(trim(DivId),trim(ClId)) = TfClasses desc , left(TfClasses,1)!='_' and left(TfClasses,1)!='%' desc , left(TfClasses,1)='_' desc , TfClasses desc , TfClasses='%'"
#1: Unrecognized keyword. (near "and" at position 959)
#2: Unrecognized keyword. (near "left" at position 963)
#3: Unexpected token. (near "(" at position 967)
#4: Unexpected token. (near "TfClasses" at position 968)
#5: Unexpected token. (near "," at position 977)
#6: Unexpected token. (near "1" at position 978)
#7: Unexpected token. (near ")" at position 979)
#8: Unexpected token. (near "!=" at position 980)
#9: Unexpected token. (near "'%'" at position 982)
#10: A new statement was found, but no delimiter between it and the previous one. (near "desc" at position 986)
```

Simplified testcase:

```
./bin/lint-query --query 'SELECT 1 ORDER BY 1 AND 1'
#1: Unrecognized keyword. (near "AND" at position 20)
#2: Unexpected token. (near "1" at position 24)
```

**Comment by nijel** [comment]
The problem here is that we do not understand string operators at all:

```
$ ./bin/lint-query --query 'SELECT 1 AND 1'
#1: Unrecognized keyword. (near "AND" at position 9)
#2: Unexpected token. (near "1" at position 13)
$ ./bin/lint-query --query 'SELECT 1 DIV 1'
#1: Unrecognized keyword. (near "DIV" at position 9)
#2: Unexpected token. (near "1" at position 13)
```

**Comment by nijel** [comment]
See also http://dev.mysql.com/doc/refman/5.7/en/non-typed-operators.html

**Comment by nijel** [comment]
Fixed in 6bb07eaa6672ae56f02e24ec449dfa71d456385b.
